lib/views/home/home_view.dart

```diff
@@ -55,39 +57,47 @@ class HomeViewState extends ConsumerState<HomeView> {
         },
       );
 
-  Widget get _surveySection => Consumer(builder: (_, ref, __) {
-        final surveyList = ref.watch(surveyListStream).value ?? [];
-        return Column(
-          mainAxisAlignment: MainAxisAlignment.start,
-          crossAxisAlignment: CrossAxisAlignment.start,
-          children: [
-            SurveyPageIndicator(
-              controller: _surveyItemController,
-              count: surveyList.length,
-            ),
-            SingleChildScrollView(
-              child: SizedBox(
-                height: 160,
-                child: SurveyList(
-                  surveyList: surveyList,
-                  itemController: _surveyItemController,
-                  onItemChange: _surveyIndex,
-                ),
-              ),
-            ),
-          ],
-        );
-      });
+  Widget _surveyList(List<Survey> surveyList) => SurveyList(
+        refreshStyle: RefreshStyle.pullDownToRefresh,
+        surveyList: surveyList,
+        itemController: _surveyItemController,
+        onItemChange: _surveyIndex,
+        onRefresh: () => _fetchSurveyList(isRefresh: true),
+        onLoadMore: () => _fetchSurveyList(isRefresh: false),
+      );
 
-  Widget get _mainBody => Column(
-        mainAxisAlignment: MainAxisAlignment.spaceBetween,
-        crossAxisAlignment: CrossAxisAlignment.stretch,
+  Widget _pageIndicatorSection(int length) => Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
         children: [
-          _homeHeader,
-          _surveySection,
+          const Spacer(),
+          SurveyPageIndicator(
+            controller: _surveyItemController,
+            count: length,
+          ),
+          const SizedBox(height: 160),
         ],
       );
 
+  Widget get _mainBody => Consumer(
+        builder: (_, ref, __) {
+          final surveyList = ref.watch(surveyListStream).value ?? [];
+          final index = ref.read(focusedItemIndexStream).value ?? 0;
```

Comment:
same here, please change to `watch`

Reply:
@manh-t 

I want to rebuild the SurveyList whenever `surveysStream` gets a new value. If I change to `watch`, which means the SurveyList will be rebuilt when `surveysStream` or `focusedItemIndexStream` get new values, but it doesn't work that way.

I will move it into the Future callback block, as you mentioned in the chat thread.